src/dbnode/storage/shard.go

```diff
@@ -1937,6 +1951,23 @@ func (s *dbShard) UpdateFlushStates() {
 	readInfoFilesResults := fs.ReadInfoFiles(fsOpts.FilePathPrefix(), s.namespace.ID(), s.shard,
 		fsOpts.InfoReaderBufferSize(), fsOpts.DecodingOptions(), persist.FileSetFlushType)
 
+	// TODO: use for blockSize differences
+	blockSize := s.namespace.Options().RetentionOptions().BlockSize()
+	indexBlockSize := s.namespace.Options().IndexOptions().BlockSize()
+
+	// expose for getting all info files
+	indexFlushedBlockStarts := s.reverseIndex.WarmFlushedBlockStarts()
+	for _, blockStart := range indexFlushedBlockStarts {
+		// Index block size is wider than data block size, so we want to set all data blockStarts
+		// within the range of a given index blockStart
+		for at := blockStart; at < blockStart.Add(indexBlockSize); at = at.Add(blockSize) {
+			currState := s.flushStateNoBootstrapCheck(at)
+			if currState.WarmStatus.DataFlushed != fileOpSuccess {
+				s.markWarmIndexFlushStateSuccess(at)
```

Comment:
You don't do this when `DataFlushed == fileOpSuccess` because that would mean we've already perform `markWarmIndexFlushStateSuccess`, yes?

Reply:
Ah this is a bug. Should be checking `IndexFlushed`